src/Builder/FrontendDialectTransformer.cpp

```diff
@@ -66,16 +70,50 @@ class FrontendGenImpl {
       const onnx::NodeProto &);
 
   std::map<std::string, ImportHandlerType> import_handler_map_;
+  std::map<std::string, int64_t> opset_map_;
+  std::map<std::string, onnx::ValueInfoProto> value_info_map;
 
   mlir::Location UnknownLoc() { return mlir::UnknownLoc::get(&context_); }
 
+  void AddValueInfo(const onnx::ValueInfoProto &vi) {
+    value_info_map[vi.name()] = vi;
+  }
+
+  void SetOpSetImport(const onnx::ModelProto &model) {
+    opset_map_.clear();
+    for (auto &binding : model.opset_import()) {
+      opset_map_[binding.domain()] = binding.version();
+    }
+  }
+
+  // TODO: Update ImportInputTensorSymbol to call this
+  void BindOnnxName(const std::string &onnx_name, mlir::Value symbol) {
```

Comment:
@chentong319 don't we have a similar mechanism in place already? Could this part use it instead of re-creating it?

Reply:
Yes, but there is a minor difference (the other method takes a ValueInfo as a parameter). I have updated the other method to call this (more primitive method).